src/ethereum/frontier/spec.py

```diff
@@ -51,6 +51,45 @@ class BlockChain:
     state: State
 
 
+def get_recent_block_hashes(
+    chain: BlockChain, num_blocks: Uint
+) -> List[Hash32]:
+    """
+    Obtain the list of hashes of the previous `num_blocks` blocks in the
+    order of increasing block number.
+
+    Parameters
+    ----------
+    chain :
+        History and current state.
+    num_blocks :
+        Number of recent block hashes one wishes to obtain.
+
+    Returns
+    -------
+    recent_block_hashes : `List[Hash32]`
+        Hashes of the recent `num_blocks` blocks in order of increasing
+        block number.
+    """
+    if len(chain.blocks) == 0:
+        return []
+
+    # We are computing the hash only for the most recent block and not for
+    # the rest of the blocks as they have successors which have the hash of
+    # the current block as parent hash.
+    most_recent_block_hash = crypto.keccak256(rlp.encode(chain.blocks[-1]))
+    recent_block_hashes = [most_recent_block_hash]
+
+    for idx, block in enumerate(reversed(chain.blocks)):
+        if idx == num_blocks - 1:
+            break
```

Comment:
Hi Sam. I don' think I understood this perfectly. I think we do need `enumerate` to stop when we have obtained `n` block hashes. Do you think we can do that without using `enumerate` as well? Thanks.

Reply:
> Hi Sam. I don' think I understood this perfectly. I think we do need `enumerate` to stop when we have obtained `n` block hashes. Do you think we can do that without using `enumerate` as well? Thanks.

Yep! You can shorten the length of the array with the `:` in the array subscript. In the suggestion block, I skip the first item with `[1:]`.

Reply:
Actually, the `reverse` operator is giving a generator instead of a list, hence not being `indexable` or `sliceable` unless it's converted to list. I can convert to list as well, but the below workflow seems efficient and readable to me. Nevertheless let me know if it's ok to convert the generator to list, over the below solution.
```
recent_block_hashes = [most_recent_block_hash]
for block in reversed(chain.blocks):
    if len(recent_block_hashes) == num_blocks:
        break

    prev_block_hash = block.header.parent_hash
    recent_block_hashes.append(prev_block_hash)
```

Reply:
I've actually implemented the above in the latest changes. Please let me know if you see any issues. Thanks.

Reply:
> Actually, the `reverse` operator is giving a generator instead of a list, hence not being `indexable` or `sliceable` unless it's converted to list.

Right, but you can slice off the first item of `chain.blocks` to remove the last item of `reversed(chain.blocks)`, no?

Something like `reverse(chain.blocks[1:num_blocks])`?

Reply:
@SamWilsn I have implemented the above suggestion. But I did it with negative indexing, which is also what you meant I assume. Please let me know how it looks or please correct me if I am still missing something. Thanks.